commune.lic

```diff
@@ -0,0 +1,412 @@
+	when 'You put'
+	  return true
+	else
+	  return false
+	end
+  end
+
+  def empty_cleric_hands(hand='both')
+    if hand == 'both'
+	  empty_cleric_right_hand
+	  empty_cleric_left_hand
+	elsif hand == 'right'
+	  empty_cleric_right_hand
+	elsif hand == 'left'
+	  empty_cleric_left_hand
+	end
+  end
+  
+  def empty_cleric_right_hand()
+  	 stow_cleric_item(GameObj.right_hand.noun) if @cleric_items.include?(GameObj.right_hand.name)
+	 stow_cleric_item(GameObj.right_hand.noun) if @cleric_items.include?(GameObj.right_hand.noun)
+	 fput('stow right') if GameObj.right_hand.noun != nil
+  end
+  
+  def empty_cleric_left_hand()
+  	 stow_cleric_item(GameObj.left_hand.noun) if @cleric_items.include?(GameObj.left_hand.name)
+	 stow_cleric_item(GameObj.left_hand.noun) if @cleric_items.include?(GameObj.left_hand.noun)
+	 fput('stow left') if GameObj.left_hand.noun != nil
+  end
+  
+  ######################## ITEM CHECKS ########################
+
+  def have_holy_water?()
+	case bput("count holy water in my #{@water_holder}", 'left of the holy water', 'I could not find', 'What were you referring to')
+	when 'left of the holy water'
+	  return true
+	else
+	  echo('No holy water found. Please make sure containers are open.')
+	  missing_item('holy water')
+	end
+  end
+
+  def have_cleric_item?(item)
+    case bput("look at my #{item} in my #{@theurgy_supply_container}", 'You see nothing unusual', 'I could not find', 'What were you referring to')
+    when 'You see nothing unusual'
+	  return true
+    else
+	  echo("No #{item} found. Please make sure containers are open.")
+	  missing_item("#{item}")
+    end
+  end
+  
+  def holding?(noun)
+    if GameObj.right_hand.noun == "#{noun}"
+	  return true
+	elsif GameObj.left_hand.noun == "#{noun}"
+	  return true
+	else
+	  return false
+	end
+  end
+  
+  # must pass items in an array
+  def have_cleric_items?(items)
+    items.each do |item|
+	  if
+	    item == 'holy water' then have_holy_water?
+	  elsif
+	    @cleric_items.include?("#{item}") then have_cleric_item?("#{item}")
+	  else
+	    echo("Unrecognized item required: #{item}")
+	    exit
+	  end
+	end
+	return true
+  end
+  
+  def missing_item(item)
+    echo("### You do not have #{item}. Exiting commmune script. ###")
+    exit
+  end
+
+  ######################## COMMUNE ACTIONS ########################
+  
+  def sprinkle_holy_water(target=@self)
+	get_cleric_item(@water_holder)
+    sprinkle(@water_holder, target)
+	stow_cleric_item(@water_holder)
+  end
+  
+  def sprinkle_holy_oil(target=@self)
+    get_cleric_item('holy oil')
+    sprinkle('oil', target)
+    stow_cleric_item('holy oil') if holding?('oil')
+  end
+  
+  def sprinkle(item, target=@self)
+    case bput("sprinkle my #{item} on #{target}", 'You sprinkle', 'Sprinkle what', 'What were you referring to')
+	when 'You sprinkle'
+	  return true
+	else
+	  return false
+	end
+  end
+  
+  def wave_incense(target=@self)
```

Comment:
In case a future use of the script requires waving at something else. It could be eliminated in the current version of the script with no change in functionality.